utils/filesystem.go

```diff
@@ -183,5 +189,69 @@ func UnZip(zipFileName, fileDestination string) {
 			errors.CheckErr(err, 404, "")
 		}
 	}
-	log.Println("File extracted:", zipFileName)
+	log.Printf("Extracted file: from '%s' to '%s'\n", filePath, destination)
+	return nil
+}
+
+// UnTar unpacks a tar.gz file to a destination
+func UnTar(pathToTarFile, destination string) error {
+	fileReader, err := read(pathToTarFile)
+	if err != nil {
+		return err
+	}
+	defer fileReader.Close()
+	gzipReader, err := gzip.NewReader(fileReader)
+	if err != nil {
+		return err
+	}
+	defer gzipReader.Close()
+	tarReader := tar.NewReader(gzipReader)
+	for {
+		header, err := tarReader.Next()
+		if err == io.EOF {
+			// end of tar archive
+			break
+		}
+		if err != nil {
+			panic(err)
```

Comment:
Is there a reason for using a `panic` here? This will print the entire stack trace (including build path) which is very verbose. I think `log.fatal` is enough information to print out and will automatically be followed by `os.Exit(1)`. - this could also be moved to the error module